src/test/java/uk/gov/cshr/controller/MaintenanceControllerDisabledTest.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.cshr.controller;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import uk.gov.cshr.AppInsightsHelper.Helper;
+
+import javax.servlet.http.Cookie;
+import javax.transaction.Transactional;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.redirectedUrl;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@Transactional
+@TestPropertySource(properties = {"MAINTENANCE_ENABLED = false"})
+public class MaintenanceControllerDisabledTest {
```

Comment:
not sure if adding another test scenario `shouldNotRedirectToMaintenanceWhenCallingOauth` is an overkill?

Reply:
can remove